pwncat/platform/linux.py

```diff
@@ -494,36 +494,65 @@ class LinuxPath(pathlib.PurePosixPath):
     def readable(self):
         """Test if a file is readable"""
 
-        uid = self._target._id["euid"]
-        gid = self._target._id["egid"]
+        # refresh the uid, to pick up any changes
+        self._target.refresh_uid()
+
+        uid = self._target._id["ruid"]
+        gid = self._target._id["rgid"]
+        euid = self._target._id["euid"]
+        egid = self._target._id["egid"]
         groups = self._target._id["groups"]
 
-        file_uid = self.stat().st_uid
-        file_gid = self.stat().st_gid
-        file_mode = self.stat().st_mode
+        # get the stats for the current path
+        _stat = self.stat()
```

Comment:
The results of `stat` are cached in the path object, but I'm fine with doing it this way. :shrug: no sweat off my back.

Reply:
I just wanted to note that the results are cached, so don't be afraid to call it multiple times. Doesn't require a discussion; just a side-note, so I'm resolving this comment immediately. :+1:

Reply:
Unfortunately, those are not __cached__.
I could see the commands being executed __thrice__ in a row using _live input/output logs_ via `MITM`

Reply:
**Uploading a file**
![Screenshot from 2021-08-11 00-46-40](https://user-images.githubusercontent.com/44923300/128922116-e77f59c3-1be6-4850-901d-bef0526b2464.png)

**BTS* traffic**
![Screenshot from 2021-08-11 00-50-58](https://user-images.githubusercontent.com/44923300/128922107-fcf38ebb-309c-4c09-a7e4-372d80b67d6b.png)

Here, 4 `stat` calls are made. 3 out of which are for the _target file_

Reply:
You are correct, those are actually saved into `_stat`, I must have missed something.
I think, I surely need some sleep :joy: